app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -72,11 +72,10 @@ def send_confirmation_code
     # user's session. Re-sending the confirmation code doesn't generate a new one.
     self.confirmation_code = generate_confirmation_code unless confirmation_code
 
-    if current_otp_method == :voice
-      VoiceSenderOtpJob.perform_later(confirmation_code, unconfirmed_phone)
-    else
-      SmsSenderOtpJob.perform_later(confirmation_code, unconfirmed_phone)
-    end
+    job = "#{current_otp_method.to_s.capitalize}SenderOtpJob".constantize
+
+    job.perform_later(confirmation_code, unconfirmed_phone)
+
```

Comment:
This is a common way of implementing the "Replace Conditional with Polymorphism" refactoring pattern. At least here, you can grep for most of the class name, the `SenderOtpJob` portion, which is common to all the classes related to sending an OTP, because we have agreed to this naming convention. 

If Sandi Metz uses this pattern, it makes it acceptable in my opinion. Search for `const_get` in this article:

http://www.sandimetz.com/blog/2014/9/9/shape-at-the-bottom-of-all-things

thoughtbot, a well-respected Ruby/Rails firm, also uses this pattern on page 45 of their [Ruby Science](https://www.scribd.com/document/266755527/thoughtbot-Joe-Ferris-Harlow-Ward-Ruby-Science-pdf) book.

The other option would be to create a Class or Module that defines a hash that maps the delivery method as submitted by the user to the Active Job class, and then the class can be obtained from the hash. This would not be my first choice though because you have to remember to update the hash every time you add a new delivery method and corresponding Active Job class.